libbeat/docs/processors-list.asciidoc

```diff
@@ -0,0 +1,74 @@
+// TODO: Create script that generates this file. Condiitional coding needs to
+// be preserved.
+
+//# tag::processors-list[]
+* <<add-cloud-metadata,`add_cloud_metadata`>>
+* <<add-docker-metadata,`add_docker_metadata`>>
+* <<add-fields, `add_fields`>>
+* <<add-host-metadata,`add_host_metadata`>>
+* <<add-kubernetes-metadata,`add_kubernetes_metadata`>>
+* <<add-labels, `add_labels`>>
+* <<add-locale,`add_locale`>>
+* <<add-observer-metadata,`add_observer_metadata`>>
+* <<add-process-metadata,`add_process_metadata`>>
+* <<add-tags, `add_tags`>>
+* <<community-id,`community_id`>>
+* <<convert,`convert`>>
+* <<decode-base64-field,`decode_base64_field`>>
+ifdef::has_decode_cef_processor[]
+* <<processor-decode-cef,`decode_cef`>>
+endif::[]
+ifdef::has_decode_csv_fields_processor[]
```

Comment:
generating + selective attributes might be painful.

How about always creating a negating conditional, but have a "no" attribute like:

```
ifndef::no_processor_decode_csv_fields
...
endif::[]
```

In this case beats have to opt-out. But we already have a very selective opt-in. Maybe selective opt-out, with support for every processor type might be more easy to implement (just an idea).

Reply:
That's a good point. When I created the conditions, I did whatever approach meant the fewest touched files. :-) But opt-out might be easier because most Beats support all processors.

Reply:
I hesitate to do any more big changes before merging this PR, so I've created an issue to track the additional work: https://github.com/elastic/beats/issues/14490. To avoid making this PR more complex I'll defer the conditional coding changes until after this is merged.